pkg/service/did/ion.go

```diff
@@ -88,6 +91,287 @@ func (i ionStoredDID) IsSoftDeleted() bool {
 	return i.SoftDeleted
 }
 
+type UpdateDIDRequest struct {
+	DID ion.ION `json:"did"`
+
+	StateChange ion.StateChange `json:"stateChange"`
+}
+
+type UpdateDIDResponse struct {
+	DID did.Document `json:"did"`
+}
+
+type preAnchor struct {
+	updateOperation      *ion.UpdateRequest
+	nextUpdatePublicJWK  *jwx.PublicKeyJWK
+	nextUpdatePrivateJWK *jwx.PrivateKeyJWK
+	storedDID            *ionStoredDID
+}
+
+type anchor struct {
+	// The result of calling anchor.
+	err error
+}
+
+type updateState struct {
+	status    UpdateRequestStatus
+	preAnchor *preAnchor
+	anchor    *anchor
+}
+
+func (h *ionHandler) UpdateDID(ctx context.Context, request UpdateDIDRequest) (*UpdateDIDResponse, error) {
+	updateStates, updatePrivateKey, err := h.readUpdateState(ctx, request.DID.String())
+	if err != nil {
+		return nil, err
+	}
+	state := &updateStates[len(updateStates)-1]
+	if state.status == DoneStatus {
+		updateStates = append(updateStates, updateState{})
+		state = &updateStates[len(updateStates)-1]
+	}
+
+	watchKeys := []storage.WatchKey{
+		{
+			Namespace: updateRequestStatusNamespace,
+			Key:       request.DID.String(),
+		},
+	}
+
+	if state.status == "" {
+		execResp, err := h.storage.db.Execute(ctx, func(ctx context.Context, tx storage.Tx) (any, error) {
+			didSuffix, err := request.DID.Suffix()
+			if err != nil {
+				return nil, errors.Wrap(err, "getting did suffix")
+			}
+
+			updateKey := updatePrivateKey.ToPublicKeyJWK()
+
+			signer, err := ion.NewBTCSignerVerifier(*updatePrivateKey)
```

Comment:
why not call https://github.com/TBD54566975/ssi-sdk/blob/main/did/ion/operations.go#L175 instead

Reply:
I wasn't able to find a way in which I could create an instance of `ion.DID` with a custom update key. The only way to create an instance of that struct is by calling [NewIONDID](https://github.com/TBD54566975/ssi-sdk/blob/9617360e9ee224b9ebda63eb52c214bc771e7cfa/did/ion/operations.go#L123), which creates new update keys. 

I agree that it would be desirable, but didn't want to block this PR on an SDK update.

Reply:
ok fair